### ISSUE TITLE
Audience distribution: Hugging Face publish, install verification, community post, intent guides

### ISSUE THREAD

**Issue by sarthakagrawal927** [PostTrainLLM/posttrainllm#148]
## Why

PostTrainLLM already qualifies as a shareable browser learning experiment (fleet shareability audit, 2026-09-07) and is the only fleet product that won a real category SERP ("fine-tune LLM specialists on Mac Apple Silicon", class A). But its actual audience — Mac-local LLM practitioners — lives on Hugging Face and r/LocalLLaMA, and the project has **zero Hugging Face presence** (`author:PostTrainLLM` returns no models) despite having trained specialist artifacts in `specialists/registry.json` (Pace intent router 49.5M/95.5%/3ms, file-ops distilled 4B at 100% hard gate, ReST fused 4B).

The strongest shareable result is already written up: a 4B model distilled locally from ~99 frontier rollouts matches DeepSeek-V4-pro at 100% on a multi-turn agentic tool-calling gate (`docs/learn/tool-calling-frontier-parity.md`). It has never been put in front of the audience it was made for.

This issue is deliberately scoped under the fresh-experiment gate: it authorizes **distribution work only** — no new training runs, no new experiments.

## What

**In scope**
- Hugging Face org/account for PostTrainLLM and publication of 2–3 specialist artifacts with complete model cards (metrics, intended use, license, links back to posttrainllm.com + this repo).
- Fresh-install verification receipt: clean clone → documented quickstart → one train/sample/export run, closing the Sep-7 native-claims gate.
- One findings-first community post draft (r/LocalLLaMA primary; Show HN optional) built on the frontier-parity result and the 0.6B≈4B distillation win.
- Recipe→query map: match the 18 recipes to real search queries and publish/refresh the top-3 intent guides.

**Out of scope**
- New experiments or training runs (fresh-experiment gate applies — this is a distribution issue, not a factory run).
- Browser/WebGPU polish, Astro migration, Mac app polish, ANE work — all remain parked.
- Commercial positioning or monetization framing; this stays a learning-lab share.

## Design

- **HF publish:** pick the strongest 2–3 artifacts from `specialists/` (candidates: Pace intent router 49.5M, file-ops distilled 4B, ReST fused 4B). Export to HF-native format where needed, write model cards carrying measured metrics + links to the matching recipe page and `docs/learn/tool-calling-frontier-parity.md`. Optional: one dataset/eval-corpus card.
- **Install gate:** fresh `git clone` on a clean user/machine → README Quickstart → `train --preset tiny` → `sample` → `export-mlx`; paste output into this issue as the receipt.
- **Community post:** findings-first framing — the numbers ARE the post. "A 4B trained on one Mac matched DeepSeek-V4-pro on agentic tool-calling; here's the recipe, the eval gate, and the honest failures." Links to docs, not marketing.
- **Intent guides:** run seo-research mapping over all 18 recipes; publish/refresh the 3 highest-demand guides on posttrainllm.com (the `/mlx-lora-fine-tuning` pattern that already won a SERP).

### Requirement: Hugging Face presence
The project SHALL publish at least two specialist artifacts on Hugging Face with complete model cards.

#### Scenario: Practitioner discovery
- **WHEN** a user searches HF for Mac/MLX fine-tuned specialists
- **THEN** they find a PostTrainLLM artifact whose card states measured metrics, license, and links back to the recipe and repo

### Requirement: Install verification
A clean-machine run of the documented quickstart SHALL be completed and recorded in this issue.

#### Scenario: Fresh clone repro
- **WHEN** a fresh clone follows the README quickstart
- **THEN** train (tiny preset), sample, and export-mlx each complete and the receipt is pasted here

### Requirement: Community post
The project SHALL produce one findings-first post draft for r/LocalLLaMA, owner-reviewed before posting.

#### Scenario: Findings-first framing
- **WHEN** the draft is reviewed
- **THEN** it leads with measured results and honest failures, links to docs — not landing-page copy

### Requirement: Intent guides
The project SHOULD map all 18 recipes to real search queries and publish or refresh at least 3 guides targeting the highest-demand queries.

### Requirement: Boundary
This issue SHALL NOT authorize new experiments, training runs, or parked-lane work (browser polish, Astro, ANE, app polish).

## Tasks

- [ ] 1. Create HF org/account; select 2–3 specialist artifacts from `specialists/registry.json`
- [ ] 2. Export/package artifacts for HF; write model cards with metrics, license, backlinks
- [ ] 3. Fresh-clone install verification: quickstart → tiny train → sample → export-mlx; paste receipt here
- [ ] 4. Draft r/LocalLLaMA post on frontier-parity + distillation results; owner reviews
- [ ] 5. Map 18 recipes to search queries (seo-research); pick top 3
- [ ] 6. Write/refresh the 3 intent guides on posttrainllm.com
- [ ] 7. Post HF links + community post; record dates/links here for geo-observatory tracking

**Comment by sarthakagrawal927** [comment]
Distribution checkpoint (2026-09-19): the zero-Hugging-Face premise was stale. All three target repositories already contain public artifacts; no signup or weight upload was necessary. Corrected and live-verified their README model cards: pace-intent-router-v8 commit 3744b582f8c048d6d4e9cbe788c4a55ca5803847; qwen3-4b-file-ops-distilled commit 957e6ad7eb3654166f094f8467e9f78284469a62; qwen3-4b-rest-fused commit 05f7b4842c1d1e399e44283462b2d299d891020b. Cards now link measured evidence and the Mac quickstart; router synthetic success is not presented as a production win and ReST breadth regression is explicit. Source synchronization is PR #149. Complete licensing/provenance qualification, clean-machine quickstart receipt, recipe query mapping/guides, and owner-reviewed community publication remain open. No new experiment or community post was run/sent.

**Comment by sarthakagrawal927** [comment]
Merged #149 at d9aa2c11d90eb7e41fc4b56030a26544d47e1998 after all eight hosted checks and independent claim/hash review passed. Three existing Hugging Face model cards were corrected and verified publicly; source includes the 18-recipe query map, three refreshed intent guides, and an explicitly unapproved community draft. No new model experiment or community submission was performed. Still open: specialist license/provenance completion; clean-machine native quickstart receipt (requires the repo-required approval before a sustained build/training loop); owner review and publication of the community draft; final live-guide deployment verification. Source merge is not proof of production deployment. The existing awesome-mlx upstream PR #28 remains an external-maintainer dependency.

**Comment by sarthakagrawal927** [comment]
## Concise PRD 2 — PostTrainLLM published with honest evidence
**ROI rank:** 2/6 · **Difficulty:** 2/5; second easiest. Licensing, owner review and tool access may add waiting time.

### Why
PR #149 corrected existing Hugging Face cards and three guides. Remaining value comes from reproducibility and practitioner feedback, not more training.

### What
Complete license/provenance review, the documented clean-install check, live-guide verification and one owner-reviewed findings-first community post. Exclude new experiments, model variants, UI polish and signup flows.

### Design
Reuse existing artifacts and report hashes. Separate narrow benchmark results, negative outcomes and unknowns; do not promote rejected general-purpose claims.

### Specs
WHEN a practitioner follows the approved clean-clone tiny train/sample/export quickstart, THEN each step completes with a dated receipt. Published cards SHALL have justified license/provenance statements; unresolved rights block publication claims. No audience-growth target is a completion gate.

### Tasks
- [ ] Resolve artifact licensing and verify current cards/backlinks.
- [ ] Obtain required heavy-run approval; execute the exact clean-install smoke.
- [ ] Verify the three live guides; secure owner approval, publish once and record links.

**Models:** SWE-2 for documentation, packaging and smoke fixes. GPT-5.6 Terra medium for an independent claims-to-evidence review; Astra only for unresolved technical contradictions. Legal uncertainty requires owner/legal resolution, not a bigger model.

**Execution boundary:** Proposed PRD, not a completion claim or permission to deploy, migrate, release, post publicly, or perform browser signup. Existing excluded/deferred work is not automatically reactivated.

**Model policy:** SWE-2 is the free preferred implementer (owner-provided). Escalations below are risk-based recommendations, not measured model comparisons. OpenAI model roles: https://learn.chatgpt.com/docs/models.

**Comment by sarthakagrawal927** [comment]
Clean-install smoke receipt (PRD 2, heavy-run approved 2026-09-19).

**Environment:** macOS on Apple Silicon (M5 Pro/48 GB), Xcode 27.0 (`/Applications/Xcode-27.0.0-Beta.4.app`), fresh `git clone --depth 1` at `d9aa2c11` into `/tmp/ptllm-clean`. The `xcodebuild -downloadComponent MetalToolchain` step was already satisfied by the installed toolchain; no download was needed.

**Steps (exact README quickstart):**

1. `xcrun swift build -c release` in `native-mac/` → **Build complete in 90.90 s** (clean checkout, all SPM deps fetched fresh).
2. `posttrainllm train --preset tiny --steps 200 --corpus ../data/examples/tiny-corpus.txt --out /tmp/tiny.tinygpt` → **200 steps in 10.9 s** (18.4 step/s, GPU), loss 5.931 → **2.578**, peak 276.8 MB, checkpoint written.
3. `posttrainllm sample /tmp/tiny.tinygpt --prompt "ROMEO:" --tokens 100` → **100 tokens in 0.53 s** (190 tok/s, KV-cached). Output is expected 200-step gibberish; the step completes.
4. `posttrainllm export-mlx /tmp/tiny.tinygpt --out /tmp/tiny-mlx` → **68 tensors, 3.2 MB**, full MLX package (`model.safetensors`, tokenizer + configs, `mlx_load.py`).

**Doc bug found by the smoke (fixed in PR #150):** README said `sample --max-tokens`; the CLI only accepts `--tokens` — `--max-tokens` is rejected and prints usage.

Receipt date: 2026-09-19. No new experiment, training run beyond the documented tiny preset, or community submission was performed.

**Comment by sarthakagrawal927** [comment]
PRD 2 execution record (2026-09-19) — all gates closed except the owner's own Reddit submission.

**Done**

- **Licensing/provenance — resolved and live.** Qwen-derived cards moved `license:other` → `apache-2.0` (matches `Qwen/Qwen3-4B-Instruct-2507`); the from-scratch router stays MIT with an explicit justification. All three cards carry a License and provenance section (distilled = ~99 checker-passing DeepSeek-V4-pro rollouts + teacher-free gold-clone reproduction; ReST = teacher-free; router = fully synthetic, own-repo generators). Live HF commits: router `c21594e`, file-ops `53a2d49` + `c896140` wording pass, rest-fused `440efde` + `de98284`.
- **Independent claims review — GPT-5.6 Terra, medium, two passes.** Pass 1 (`request_changes`): fixed README baseline errors (Gemma-4-12b-qat 83% / stock+plan 75%), the unsupported 960M browser-training claim → measured 473M Memory64 allocation + ABI limit, the 96M→221M Huge decode label, unreceipted router time/size claims, missing sealed-V1 reject context. Pass 2 (`request_changes`): regenerated report-card SHA-256 receipts (`publish_report_cards.py --check` clean), reworded derived-weights provenance, marked benchmark-entry `training_time` missing, aligned registry `ToolRouterModel-medium`. Source synchronization is PR #150, merged `df5367c` with all 8 CI checks green.
- **Clean-install smoke — receipt above** (comment-5740956400): fresh clone at `d9aa2c1` → release build 90.9 s → tiny train 200 steps in 10.9 s → sample 190 tok/s → export-mlx 68 tensors. Found and fixed a real quickstart bug (`--max-tokens` → `--tokens`).
- **Live guides — verified.** All three pages were stale pre-deploy; deploy run `35437390950` shipped the Sep-19 refresh and `/mlx-lora-fine-tuning`, `/evaluate-local-llm`, `/build-small-language-model-specialist` now carry the ReST paired numbers (9/12→12/12 depth, 30/45→25/45 breadth) live.

**Remaining owner action**

- **Community post:** draft in `docs/distribution-2026-09-19.md` is owner-approved as-is; all three pre-posting gates (wording, license/provenance, install receipt) are satisfied. Posting to r/LocalLLaMA is the owner's action — record the link and date here once published.

No new experiment, model variant, signup flow, or UI polish was performed. awesome-mlx upstream PR #28 remains an external-maintainer dependency.

**Comment by sarthakagrawal927** [comment]
Owner-directed backlog cleanup: closing this broad distribution bundle as not planned in the current issue queue. PostTrainLLM remains primary; future distribution should start from one concrete campaign.
